iroh/src/magicsock/node_map.rs

```diff
@@ -123,11 +127,18 @@ pub enum Source {
 }
 
 impl NodeMap {
+    #[cfg(not(any(test, feature = "test-utils")))]
     /// Create a new [`NodeMap`] from a list of [`NodeAddr`]s.
     pub(super) fn load_from_vec(nodes: Vec<NodeAddr>) -> Self {
         Self::from_inner(NodeMapInner::load_from_vec(nodes))
     }
 
+    #[cfg(any(test, feature = "test-utils"))]
+    /// Create a new [`NodeMap`] from a list of [`NodeAddr`]s.
+    pub(super) fn load_from_vec(nodes: Vec<NodeAddr>, path_selection: PathSelection) -> Self {
```

Comment:
I would prefer to make this a seperate method, like `set_path_selection`, instead of changing the signature

Reply:
Issue is, the inner requires the flag and it needs to be provided at load time. Should I keep it or move forward and unpack the whole thing?